core/builtin.py

```diff
@@ -526,7 +531,13 @@ def Cd(argv, mem, dir_stack):
     util.error("cd %r: %s", dest_dir, os.strerror(e.errno))
     return 1
 
-  state.SetGlobalString(mem, 'PWD', dest_dir)  # Set $PWD.
+  # Set $PWD.
+  if arg.P:
+    state.SetGlobalString(mem, 'PWD', os.path.realpath(dest_dir))
+  else: # `-L` is the default behavior; no need to check it
```

Comment:
nitpick: I use two spaces before the # in an end-of-line comment.  And I don't like backticks unless they are actually rendered in a monospace font :)

Since the lint infrastructure is not in great shape, and I'm travelling now, I'll just merge this since it looks good otherwise.  But for future reference I'm pointing out minor style issues.